### ISSUE TITLE
inherit_from/inherit_gem ignores cwd when defining Exclude paths

### ISSUE THREAD

**Issue by fnando** [rubocop-hq/rubocop#7527]
tl;dr: when loading a configuration from outside the current path, any `Exclude` clause will prepend the config file's basedir, instead of honoring the current path, as `Include` does.

---

If a file `.rubocop.yml` with the following lines is used

```yaml
---

inherit_from:
  - ./some/deep/folder/structure/.rubocop.yml
```

And the inherited file looks like this

```yaml
---

Metrics/LineLength:
  Enabled: true
  Max: 30
  Exclude:
    - config/environments/production.rb
  Include:
    - lib/**/*.rb

Style/StringLiterals:
  EnforcedStyle: double_quotes
```

When I run `rubocop --show-cops Metrics/LineLength`, then I get this output:

```console
$ rubocop --show-cops Metrics/LineLength
# Supports --auto-correct
Metrics/LineLength:
  Description: Limit lines to 80 characters.
  StyleGuide: "#80-character-limits"
  Enabled: true
  VersionAdded: '0.25'
  VersionChanged: '0.68'
  AutoCorrect: false
  Max: 30
  AllowHeredoc: true
  AllowURI: true
  URISchemes:
  - http
  - https
  IgnoreCopDirectives: true
  IgnoredPatterns: []
  Exclude:
  - "/Users/fnando/Projects/personal/rubocop-bug-report/some/deep/folder/structure/config/environments/production.rb"
  Include:
  - lib/**/*.rb
```

Notice that `Exclude` prepends the basedir, while `Include` honors whatever I have defined.

--------

## Expected behavior

`Exclude` should honor whatever is defined on inherited files, just like `Include`.

## Actual behavior

`Exclude` prepends the inherited configuration's basedir.

## Steps to reproduce the problem

[rubocop-bug-report.zip](https://github.com/rubocop-hq/rubocop/files/3894022/rubocop-bug-report.zip)

## RuboCop version

```
$ rubocop -V         
0.76.0 (using Parser 2.6.5.0, running on ruby 2.6.5 x86_64-darwin19)
```

**Comment by buehmann** [comment]
There is this naming rule for config files that governs how relative paths are interpreted: 

> Path relativity
>
> In .rubocop.yml and any other configuration file beginning with .rubocop, files, and directories are specified relative to the directory where the configuration file is. In configuration files that don't begin with .rubocop, e.g. our_company_defaults.yml, paths are relative to the directory where rubocop is run.

https://docs.rubocop.org/en/stable/configuration/#includingexcluding-files

So I agree that `Exclude` and `Include` behave inconsistently in your example, but I believe it is the `Include` that is wrong because the config file you are inheriting from starts with `.rubocop`. Maybe you could try what changes when you rename the config file.
